client/src/controllers/DrilldownController.test.js

```diff
@@ -0,0 +1,90 @@
+import { Application } from '@hotwired/stimulus';
+
+import { DrilldownController } from './DrilldownController';
+
+jest.useFakeTimers();
+
+describe('DrilldownController', () => {
+    const eventNames = [
+        'w-drilldown:opened',
+        'w-drilldown:closed',
+        'w-drilldown:ready',
+        'w-drilldown:toggled',
+    ];
+
+    const events = {};
+
+    let application;
+    let errors = [];
+    const setup = async (
+        html = `
+    <section ><div data-controller="w-drilldown">
+    <button type="button" data-action="w-drilldown#toggle" data-w-drilldown-target="toggle" aria-expanded="false">
+      Toggle Menu
+    </button>
+    <span data-w-drilldown-target="count">0</span>
+    <ul data-w-drilldown-target="menu" hidden>
+     <li>ListItem</li>
+    </ul>
+  </div></section>`,
+        identifier = 'w-drilldown'
+    ) => {
+        document.body.innerHTML = `<main>${html}</main>`;
+
+        application = new Application();
+        application.handleError = (error, message) => {
+            errors.push({ error, message });
+        };
+        application.register(identifier, DrilldownController);
+        application.start();
+
+        await jest.runAllTimersAsync();
+
+        return [
+            ...document.querySelectorAll(`[data-controller~="${identifier}"]`),
+        ].map((element) =>
+            application.getControllerForElementAndIdentifier(element, identifier)
+        );
+    };
+    beforeEach(async () => {
+        await setup();
+    });
+
+    afterEach(() => {
+        application?.stop && application.stop();
+        errors = [];
+        eventNames.forEach((name) => {
+            events[name] = [];
+        });
+    });
+    // ------------------------------------
+    describe('when connecting to the DOM', () => {
+        it('opens and closes the menu when toggle is clicked', async () => {
+            const [controller] = await setup();
```

Comment:
Unless you're actually checking something with the controller, you probably can just call setup without a variable assignment. 


```suggestion
            await setup();
```